lib/iris/fileformats/netcdf.py

```diff
@@ -1958,7 +1949,11 @@ def set_packing_ncattrs(cfvar):
             # Explicitly assign the fill_value, which will be the type default
             # in the case of an unmasked array.
             if packing is None:
-                fill_value = cube.fill_value
+                if hasattr(cube.lazy_data(), 'fill_value'):
+                    fill_value = cube.lazy_data().fill_value
+                else:
+                    fill_value = None
+
```

Comment:
@lbdreyer Actually, we need to be slightly cautious how we do this ...
```python
>>> m = ma.masked_array([0,1,2,3], mask=[0,0,0,1], fill_value=123)
>>> dm = da.from_array(m, chunks=(1,), asarray=False)
>>> dm[0].compute()
0
>>> dm[-1].compute()
masked
>>> dm[:1].compute()
masked_array(data = [0],
             mask = [False],
       fill_value = 123)
```
The last example of slicing with `[:1]` gives us back a masked array with the fill-value, which is what we need